students/BrianNisonger/session9/mailroom.py

```diff
@@ -0,0 +1,197 @@
+        """ Takes a name, and optionally a donation list"""
+        self.name = name
+        self.donation = list(donation)
+
+    def add_donation(self, *values):
+        """Takes either a single donation, and appends it to the list, or appends a set of donations"""
+        for value in values:
+            self.donation.append(value)
+
+    def make_average(self, value):
+        """Averages the donations"""
+        return statistics.mean(value)
+
+    def make_total(self, value):
+        """Totals the donations"""
+        return sum(value)
+
+    def num_donations(self, value):
+        """Counts the donations"""
+        return len(value)
+
+    def make_thank_you(self):
+        """ Returns a thank you note for total donations of donor"""
+        return f'Dear {self.name}, Thank you for your donation of ${self.make_total(self.donation):.2f}. These funds help save the migratory butterflies of New South East.Thank you'
+
+    def make_filename(self):
+        """Turns a donor name into a filename"""
+        return f'{self.name.replace(" ","_")}.txt'
+
+    def make_report_string(self, value):
+        """Makes a single line of the report(Name,Total Donations,Num of Donatioms, Average Donations)"""
+        return f'{self.name:<20} ${self.make_total(value):<15.2f} {self.num_donations(value):<15} ${self.make_average(value):<15.2f}'
+
+    def __lt__(self, other):
+        """less than"""
+        return (self.make_total(self.donation) > other.make_total(
+            other.donation))
+
+    def __gt__(self, other):
+        """greater than"""
+        return (self.make_total(self.donation) < other.make_total(
+            other.donation))
+
+    def __eq__(self, other):
+        """equal"""
+        return (self.make_total(self.donation) == other.make_total(
+            other.donation))
+
+    def __str__(self):
+        """Returns name of the donor if str is called"""
+        return self.name
+
+    def sort_key(self):
+        """What to sort on"""
+        return (self.make_total(self.donation))
+
+
+class Donors():
+    """ This contains a list of classes, used to store multiple instances of the Donor class for grouping"""
+
+    def __init__(self, donor_list):
+        self.donor_list = donor_list
+
+    def append(self, cls):
+        """Adds another donor to the donor_list"""
+        self.donor_list.append(cls)
+
+    def list_donors(self):
+        """returns a list containing the names of all the donors in the donor list"""
+        return [donor.name for donor in self.donor_list]
+
+    def make_headers_string(self):
+        """Header for the report"""
+        headers = ("Donor Names", "Total Given", "Num Gifts", "Average Gifts")
+        header_string = f'{headers[0]:<20} {headers[1]:<15} {headers[2]:<15} {headers[3]:<15}'
+        return header_string
+
+    def sort_donors(self):
+        """Takes a list of donors and sorts by total donations"""
+        self.donor_list = sorted(self.donor_list)
```

Comment:
so that's what the comparison operators are for -- nice!

though you defined a sort_key method -- why not use that?

Reply:
and if you are going to replace it with the sorted version -- you can use .sort() 👍 

``self.donor_list.sort()``